tamr_unify_client/dataset/resource.py

```diff
@@ -123,6 +123,16 @@ def delete_records_by_id(self, record_ids):
         updates = ({"action": "DELETE", "recordId": rid} for rid in record_ids)
         return self._update_records(updates)
 
+    def delete_all_records(self):
```

Comment:
The API models this as a `DELETE` on the `/records` subcollection of a dataset, so this should be `dataset.records().delete_all()` or something like that. We may need to remodel `/records` as a `RecordCollection` rather than just giving back a stream of records in order to accomplish this.

Reply:
On second thought: `/records` isn't a proper collection yet (e.g. you can't get a member of the collection via APIs), so let's keep this code as is.